### ISSUE TITLE
When using controls and change source with useState next video only audio

### ISSUE THREAD

**Issue by mikaelbartlett** [TheWidlarzGroup/react-native-video#2750]
# Bug

When enabling system controls on the video and onEnd set a new source through useState only audio but no video is available.
The video player is also frozen as there's been another player instantiated.

## Platform
* iOS

## Environment info

React native info output:

```bash
System:
    OS: macOS 12.4
    CPU: (10) arm64 Apple M1 Max
    Memory: 26.91 GB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.14.0 - ~/.nvm/versions/node/v16.14.0/bin/node
    Yarn: 1.22.18 - ~/.nvm/versions/node/v16.14.0/bin/yarn
    npm: 8.3.1 - ~/.nvm/versions/node/v16.14.0/bin/npm
    Watchman: 2022.06.27.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /Users/mikbar/.rbenv/shims/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8512546
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.12 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2
    react-native: 0.68.2 => 0.68.2
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

Library version: 5.2.0

## Steps To Reproduce

1. Add a Video where the source is updated through the useState hook and enabled controls
2. Next video in the queue won't show
...

## Expected behaviour

The next video should be visible and seekable.

## Reproducible sample code

https://github.com/mikaelbartlett/react-native-video-issue

## Video sample
No specific video. Problems with all videos tested (HLS)

**Comment by bobbychan** [comment]
I am getting the same Error with 6.0.0-alpha.1

`rctDelegate.videoPlayerViewControllerWillDismiss(playerViewController: self)
Thread 1: Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value`

**Comment by vargajacint** [comment]
It's same here. Basically the video goes to paused state. Any update?

**Comment by belghithb** [comment]
I faced the same issue, any update please?

**Comment by baselshlewett** [comment]
Tested on simulator running iPhone 13, iOS 15.5
Tested on iPhone 13 Pro, iOS 15.6

same behaviour as OP
```
System:
    OS: macOS 12.4
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Memory: 1.72 GB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.13.1/bin/yarn
    npm: 8.15.1 - ~/.nvm/versions/node/v16.13.1/bin/npm
    Watchman: 2022.03.14.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8193401
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.11 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.65.0 => 0.65.0 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```
react-native-video version ^5.2.0

If used without controls (i.e controls={false}), video switch happens smoothly.

Workaround that's working for me:
```
<Video source={{ uri: file?.uri}}
        controls={showControls}
        onLoadStart={() => setShowControls(false)}
        onLoad={() => setShowControls(true)}
        style={ { width: '100%', height: 200 } } />
```

**Comment by Md-Mudassir47** [comment]
found a quick workaround, 
```
const [controlsEnabled, setControlsEnabled] = useState(true);

  const onEnd = () => {
// reset the controls.
    setControlsEnabled(false);
    if (module === null ) {
      navigation.goBack();
    } else {
      setVideo(videoList[video.id]);
    }
  };

        <Video
          source={{
            uri: video.videoUrl.replace(/\s+/g, ' ').trim(),
          }}
          style={{
            height: height - 100,
            width: width,
            backgroundColor: COLORS.dark,
            alignItems: 'center',
          }}
          poster={'https://i.ibb.co/M9HM0dt/image-loader.png'}
          posterResizeMode={'cover'}
          onReadyForDisplay={() => {
           //Enable it again
            setControlsEnabled(true);
          }}
          controls={controlsEnabled}
          resizeMode={'cover'}
          rate={1.0}
          onEnd={onEnd}
        />
        
        ```

**Comment by freeboub** [comment]
This issue shall finally be fixed in 6.0.0-beta.7, to be retested
